crowbar_framework/app/constraints/api_constraint.rb

```diff
@@ -15,13 +15,14 @@
 #
 
 class ApiConstraint
-  attr_reader :version
+  attr_reader :major, :minor
 
   def initialize(options)
-    @version = options.fetch(:version)
+    @major = options.fetch(:major)
+    @minor = options.fetch(:minor)
   end
 
   def matches?(request)
-    request.headers.fetch(:accept) == "application/vnd.crowbar.v#{version}+json"
+    request.headers.fetch(:accept) == "application/vnd.crowbar.v#{major}.#{minor}+json"
```

Comment:
I'm not sure what you mean here. There are no variables in this regex, only named captures.
So it should be pretty constant-friendly.